library/Translate/src/Juvix/Contextify/Environment.hs

```diff
@@ -147,9 +165,62 @@ bindingForms x =
              "case",
              ":lambda-case",
              ":declaim",
-             ":lambda"
+             ":lambda",
+             ":primitive"
            ]
 
+-- should really be moved out of here
+
+-- | @namedForms@ a list of all named special forms
+namedForms :: [NameSymbol.T]
+namedForms =
+  [ "type",
+    ":open-in",
+    ":let-type",
+    ":let-match",
+    "case",
+    ":lambda-case",
+    ":declaim",
+    ":lambda",
+    ":tuple",
+    ":primitive",
+    ":progn",
+    "declare",
+    ":infix",
+    ":list",
+    ":record-no-pun",
+    ":paren",
+    ":block",
+    ":primitive"
+  ]
+
+-- | @searchAndClosureNoCtx@ like searchAndClosure but does not rely on
+-- a context, thus the open pass can't be done.
+searchAndClosureNoCtx ::
+  (HasClosure f) =>
+  -- | the atom to dispatch on
+  Sexp.Atom ->
+  -- | The sexp form in which the atom is called on
+  Sexp.T ->
+  -- | the continuation of continuing the changes
+  (Sexp.T -> f Sexp.T) ->
+  f Sexp.T
+searchAndClosureNoCtx a as cont
```

Comment:
Can we deduplicate this a little bit (the `f as cont` bits)?

Reply:
yeah I just rushed copy and pasted it cause I set it up poorly. I should have returned a function that I then applied in a more general pattern.

Reply:
Okay fixed